### PR TITLE
feat: allow using any 3.x chokidar dependencies

### LINKED ISSUE #5142: Bump chokidar
### Bug Report Checklist

- [X] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/master/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md)
- [X] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [X] I have 'smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, my usage of Mocha, or Mocha itself.
- [X] I want to provide a PR to resolve this

### Expected

works with typescript

### Actual

chokidar is incompatible to current @types/node releases

### Minimal, Reproducible Example

-

### Versions

mocha 10.4.0 
@types/node 20.12.11
node 20.12.2

### Additional Info

If https://github.com/mochajs/mocha/issues/5114 was already done this wouldn't be an issue.

I'd just bump chokidar to ^3.6.0

### PR DESCRIPTION
**Pull request by simhnna**
## PR Checklist

- [x] Addresses an existing open issue: fixes part of #5114 
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md) were taken

## Overview

Compatibility with newer @types/node releases for typescript users

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
   "dependencies": {
     "ansi-colors": "4.1.1",
     "browser-stdout": "1.3.1",
-    "chokidar": "3.5.3",
+    "chokidar": "^3.6.0",
```
Comment (legobeat) [inline package.json:56]: Mocha is pinning to exact versions in manifest by convention. While I'd also suggest reconsidering that decision, changing that probably doesn't belong in this PR @simhnna?

```suggestion
    "chokidar": "^3.6.0",
```

File: package-lock.json [hunk 2]
```diff
@@ -11,7 +11,7 @@
       "dependencies": {
         "ansi-colors": "4.1.1",
         "browser-stdout": "1.3.1",
-        "chokidar": "3.5.3",
+        "chokidar": "^3.6.0",
```
Comment (legobeat) [inline package-lock.json:14]:
```suggestion
        "chokidar": "3.6.0",
```

File: package.json (hunk 1 above)
Comment (legobeat) [reply package.json]: Sorry, I missed the context there!

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Thanks for this! Up for switching to `^` and having this target the `chokidar` part of #5114?

**Comment by JoshuaKGoldberg** [comment]
I also tried an `npm upgrade chokidar` locally and re-running tests and such. They all worked. ✔️

**Comment by JoshuaKGoldberg** [comment]
This was released in [`mocha@10.5.0`](https://www.npmjs.com/package/mocha/v/10.5.0). 🚀

**Pull request merged**